src/api/controllers/v1/iuf/sessions.go

```diff
@@ -212,20 +196,48 @@ func (u IufController) Sync(context *gin.Context) {
 		}
 	case iuf.SessionStateTransitioning, iuf.SessionStateAborted, iuf.SessionStatePaused, iuf.SessionStateDebug, iuf.SessionStateCompleted:
 		u.logger.Infof("Sync: The session %s in activity %s is in state %s and there is nothing to do", session.Name, session.ActivityRef, session.CurrentState)
-		response = iuf.SyncResponse{}
-		context.JSON(200, response)
+		context.JSON(200, iuf.SyncResponse{})
 		return
 	default:
 		err := fmt.Errorf("sync: unknown state %s for session %s in activity %s", session.CurrentState, sessionName, session.ActivityRef)
 		u.logger.Error(err)
-		context.JSON(500, utils.ResponseError{Message: err.Error()})
 
 		session.CurrentState = iuf.SessionStateDebug
 		err = u.iufService.UpdateSessionAndActivity(session, fmt.Sprintf("Unknown state %s", session.CurrentState))
-		u.logger.Error(err)
+		if err != nil {
+			context.JSON(500, utils.ResponseError{Message: err.Error()})
+			return
+		}
 
+		context.JSON(500, utils.ResponseError{Message: err.Error()})
```

Comment:
is this line a mistake? won't `err` be `nil` here?

Reply:
Oh right good catch.